File: AssetSync/src/AssetSync.Core/Models/Device.cs
namespace AssetSync.Core.Models;

/// <summary>
/// Unified device model merged from Intune, Iru, and Snipe-IT.
/// Serial number is the match key (normalized).
/// </summary>
public class Device
{
    public string NormalizedSerial { get; set; } = string.Empty;
    public string? SerialNumber { get; set; }
    public string? DeviceName { get; set; }
    public string? Model { get; set; }
    public int? SnipeItModelId { get; set; }
    public string? AssignedUserUpn { get; set; }
    public int? SnipeItAssignedUserId { get; set; }
    public string? OsVersion { get; set; }
    public string? WindowsFeatureUpdate { get; set; }
    public string? DeviceType { get; set; }
    public int? SnipeItCategoryId { get; set; }
    public string? PlatformSource { get; set; }
    public int? SnipeItAssetId { get; set; }
    public string? SnipeItAssetTag { get; set; }
    public string? AzureAdDeviceId { get; set; }
    public string? IruDeviceId { get; set; }
    public string? OperatingSystem { get; set; }
    /// <summary>Asset tag as reported by the MDM platform (Intune Notes or Kandji asset_tag field).</summary>
    public string? MdmAssetTag { get; set; }
    /// <summary>Raw Notes field from Intune — preserved so write-back can update the tag without clobbering other note content.</summary>
    public string? IntuneNotes { get; set; }
    /// <summary>Intune managed device ID (m.Id) — required for PATCH DeviceManagement.ManagedDevices[id]. Distinct from AzureAdDeviceId.</summary>
    public string? IntuneDeviceId { get; set; }
}

File: AssetSync-macOS/AssetSyncCore/Sources/AssetSyncCore/Models/Device.swift
import Foundation

/// Unified device model merged from Intune, Iru, and Snipe-IT.
/// Serial number is the match key (normalized).
public struct Device: Identifiable, Sendable {
    public var id: String { normalizedSerial }

    public var normalizedSerial: String = ""
    public var serialNumber: String?
    public var deviceName: String?
    public var model: String?
    public var snipeItModelId: Int?
    public var assignedUserUpn: String?
    public var snipeItAssignedUserId: Int?
    public var osVersion: String?
    public var windowsFeatureUpdate: String?
    public var deviceType: String?
    public var snipeItCategoryId: Int?
    public var platformSource: String?
    public var snipeItAssetId: Int?
    public var snipeItAssetTag: String?
    public var azureAdDeviceId: String?
    public var iruDeviceId: String?
    public var operatingSystem: String?
    /// Asset tag as reported by the MDM platform (Intune Notes or Iru asset_tag field).
    public var mdmAssetTag: String?
    /// Raw Notes field from Intune — preserved so write-back can update the tag without clobbering other note content.
    public var intuneNotes: String?
    /// Intune managed device ID — required for PATCH. Distinct from azureAdDeviceId.
    public var intuneDeviceId: String?

    public init() {}
}

/// Lightweight id+name pair returned by Snipe-IT lookup endpoints.
public struct SnipeItLookup: Identifiable, Sendable {
    public var id: Int
    public var name: String

    public init(id: Int, name: String) {
        self.id = id
        self.name = name
    }
}
